CORE-10180: Escape string parameters when creating json to prevent possible injection attacks

anien opened the pull request:
- Escape string parameters
- Fix most of the kdoc parameters

anien committed:
Escape string parameters when creating json

dickon commented:
Looks okay, probably needs some unit tests.

ronanbrowne commented on tools/plugins/initial-config/build.gradle:
```diff
@@ -21,6 +21,8 @@ dependencies {
     kapt "info.picocli:picocli:$picocliVersion"
 
     implementation 'net.corda:corda-config-schema'
+    implementation "com.fasterxml.jackson.core:jackson-databind:$jacksonVersion"
```
Just double-checking you need

`implementation "com.fasterxml.jackson.core:jackson-databind:$jacksonVersion"`

as far as I see you are only using 

`com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
`

anien replied:
Ah, good point. I was trying few bits and didn't check which packages I need at the end. I'll double check.

ronanbrowne reviewed (changes requested):
See dependency query

anien committed:
Remove unneeded dependency

anien committed:
Add tests, temporary fix escaping backslash if it is part of escaping double quotes

anien committed:
Fix detekt, add comment

dickon commented:
LGTM

dickon reviewed (dismissed):
LGTM

kyriathar-r3 committed:
CORE-18268 db-worker restart causes loss of REST API High Availability (#5078)

The issue was that on forced config reconciliation we would reconcile all config sections and therefore propagate them across the cluster no matter if their config value actually changed or not as part of the defaulting process. This caused components across the cluster to restart on db-worker restarts (one of which is the http server within the rest-worker which was causing its unavailability).

* Contains the above config propagation on the db-worker side. On forced reconciliation checks if after applying the defaulting process to a config value read from the DB, it changes (expands) compared to its Kafka corresponding one and only then reconciles it

jujoramos committed:
CORE-18282: Fix Flaky Tests (#5090)

Sort the returned states before building the filters to use so that the
lower bound is always older in time than the upper bound.

anien commented:
Changing base branch shows some outstanding commits that shouldn't be part of this PR. I'll create a new one.
